src/sentry/static/sentry/app/views/settings/components/forms/controls/rangeSlider.jsx

```diff
@@ -153,7 +159,13 @@ class RangeSlider extends React.Component {
 
   componentWillReceiveProps(nextProps) {
     if (typeof nextProps.value !== 'undefined') {
-      this.setState({value: nextProps.value});
+      let {allowedValues} = this.props;
+      let sliderValue = nextProps.value;
+
+      if (allowedValues && allowedValues.indexOf(sliderValue) > -1) {
+        sliderValue = allowedValues.indexOf(sliderValue);
```

Comment:
This is a bit confusing since here sliderValue replaced by the index of the slider position and is no longer the actual slider value. I'd suggest renaming it to `indexOfSliderValue` or something along those lines.

Reply:
Renaming `state.sliderValue` or the local `sliderValue` variable?

Reply:
I'd probably make a separate `state.sliderIndex` or something to avoid using `state.sliderValue` for two different concepts interchangeably. But I'd leave this up to you in any case.